src/Workspaces/Core/Portable/FindSymbols/FindReferences/Finders/AbstractReferenceFinder.cs

```diff
@@ -720,13 +721,23 @@ public override Task<ImmutableArray<Project>> DetermineProjectsToSearchAsync(ISy
                 : SpecializedTasks.EmptyImmutableArray<Project>();
         }
 
-        public override Task<ImmutableArray<Document>> DetermineDocumentsToSearchAsync(
+        public async override Task<ImmutableArray<Document>> DetermineDocumentsToSearchAsync(
             ISymbol symbol, Project project, IImmutableSet<Document> documents,
             FindReferencesSearchOptions options, CancellationToken cancellationToken)
         {
-            return symbol is TSymbol && CanFind((TSymbol)symbol)
-                ? DetermineDocumentsToSearchAsync((TSymbol)symbol, project, documents, options, cancellationToken)
-                : SpecializedTasks.EmptyImmutableArray<Document>();
+            if (symbol is TSymbol targetSymbol && CanFind(targetSymbol))
+            {
+                var documentsToSearch = await DetermineDocumentsToSearchAsync(targetSymbol, project, documents, options, cancellationToken).ConfigureAwait(false);
+
+                if (options.IgnoreUnchangeableDocuments)
+                {
+                    documentsToSearch = documentsToSearch.WhereAsArray(d => d.CanApplyChange());
+                }
+
+                return documentsToSearch;
+            }
+
+            return ImmutableArray<Document>.Empty;
```

Comment:
we shouldn't be changing the documentstosearch.  if we are oing to have in this layer (which i don't like) we should just be filtering out the results that are in docs that you can't apply changes to.  This can just be a method (or extension) on the find-refs-result you get back.

Reply:
Reverted. See comment above